### PR TITLE
Update api documents

### PR DESCRIPTION
**Pull request by loay**
### Description

Update api documents/ js docs and address all REVIEW and FIXME tags
#### Related issues

connect to https://github.com/strongloop-internal/scrum-apex/issues/175

### Checklist

- [ ] New tests added or existing tests modified to cover all changes N/A
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: lib/datasource.js [hunk 1]
```diff
@@ -1835,9 +1834,7 @@ DataSource.prototype.log = function(sql, t) {
 
 /**
  * Freeze dataSource. Behavior depends on connector
- * REVIEW: Raymond, can you give example of what behavior(s) are typically
- * expected here? I understand that it's connector-specific, but surely there's
- * a common goal here.
+ * To continouly add artifacts to datasource untill it is frozen, but it is not really used in loopback.
```
Comment (0candy) [inline lib/datasource.js]: typo:
`continuously` and `until`

File: lib/datasource.js [hunk 2]
```diff
@@ -2197,9 +2195,8 @@ DataSource.prototype.isRelational = function() {
 /**
  * Check if the data source is ready.
  * Returns a Boolean value.
- * REVIEW: What is "obj" supposed to represent?
- * @param {Object} obj  ?
- * @param {Object} args  ?
+ * @param {Object} obj  Defered method call if the connector is not fully connected yet
```
Comment (0candy) [inline lib/datasource.js]: typo `deferred`

File: lib/datasource.js [hunk 3]
```diff
@@ -2286,10 +2283,10 @@ DataSource.prototype.ping = function(cb) {
   }
 };
 
+/*! The hidden property call is too expensive so it is not used that much
 /**
  * Define a hidden property
- * REVIEW: Why is a hidden property not writable or enumerable? What
- * purpose does it serve?
+ * It is an ulitlity to define a property to the Object with info flags
```
Comment (0candy) [inline lib/datasource.js]: Should be `a utility`

File: lib/datasource.js [hunk 4]
```diff
@@ -806,14 +804,15 @@ DataSource.prototype.supportTypes = function(types) {
   }
 };
 
+/*! In future versions, this will not maintain a strict 1:1 relationship between datasources and model classes
+/*! Moving forward, we will allow a model to be attached to a datsource. the model itself becomes a template.
```
Comment (0candy) [inline lib/datasource.js]: `datasource` and `The`

File: lib/datasource.js [hunk 5]
```diff
@@ -744,19 +744,17 @@ DataSource.prototype.mixin = function(ModelCtor) {
...
- *
- * FIXME: Deprecation notice for 4.x?
  * See [ModelBuilder.getModel](http://apidocs.strongloop.com/loopback-datasource-juggler/#modelbuilder-prototype-getmodel)
  * for details.
  */
 DataSource.prototype.getModel = function(name, forceCreate) {
   return this.modelBuilder.getModel(name, forceCreate);
 };
 
+/*! Method will be deprecated in LoopBack.next
```
Comment (raymondfeng) [inline lib/datasource.js]: Is this even valid?
```
/*!
/**
...
*/
```

**Pull request merged**